packages/jmix-react-ui/src/ui/FormWizard/useEntityEditorFormWizard.tsx

```diff
@@ -0,0 +1,82 @@
+import {
+    HasId,
+    useMetadata,
+} from "@haulmont/jmix-react-core";
+import { useEntityEditor, EntityEditorHookOptions, EntityEditorHookResult } from "../../crud/editor/useEntityEditor";
+import { jmixFront_to_ant } from "../../formatters/jmixFront_to_ant";
+import { useFormWizard } from "./FormWizardContext";
+import {useCallback, useEffect} from 'react';
+
+export interface EntityEditorFormWizardHookOptions<TEntity, TData, TQueryVars, TMutationVars>
+extends EntityEditorHookOptions<TEntity, TData, TQueryVars, TMutationVars> {}
+
+export interface EntityEditorFromWizardHookResult<TEntity, TData, TQueryVars, TMutationVars>
+extends EntityEditorHookResult<TEntity, TData, TQueryVars, TMutationVars> {
+    handleSelectStep: (stepIndex: number) => void;
+    handleNextStep: () => void;
+    handlePreviousStep: () => void;
+    handleSubmitBtn: () => void;
+}
+
+export function useEntityEditorFromWizard<
+    TEntity = unknown,
+    TData extends Record<string, any> = Record<string, any>,
+    TQueryVars extends HasId = HasId,
+    TMutationVars = unknown
+>(
+    options: EntityEditorFormWizardHookOptions<TEntity, TData, TQueryVars, TMutationVars>
+): EntityEditorFromWizardHookResult<TEntity, TData, TQueryVars, TMutationVars> {
+    const metadata = useMetadata();
+    const entityEditorData = useEntityEditor(options);
+    const {formWizardStore, formWizardHelpersRef} = useFormWizard();
+
+    useEffect(() => {
+        formWizardStore.setServerValidationErrors(entityEditorData.serverValidationErrors)
+    }, [formWizardStore, entityEditorData.serverValidationErrors]);
+
+    useEffect(() => {
+        if (entityEditorData.item) {
+            formWizardStore.setValues(
+                jmixFront_to_ant(entityEditorData.item, options.entityName, metadata)
+            );
+        }
+    }, [formWizardStore, metadata, entityEditorData.item, options.entityName]);
+
+    const handleSelectStep = useCallback(async newStepIndex => {
+        await formWizardHelpersRef.current.validateFields();
+        formWizardStore.setStepIndex(newStepIndex);
+    }, [formWizardStore, formWizardHelpersRef.current.validateFields]);
+    
+    const handleNextStep = useCallback(async () => {
+        await formWizardHelpersRef.current.validateFields();
+        formWizardStore.next();
+    }, [formWizardStore, formWizardHelpersRef.current, formWizardHelpersRef.current.validateFields]);
+
```

Comment:
I think we should not proceed to next step if there are validation errors in the current step.